packages/dropdowns/src/elements/Dropdown/Dropdown.tsx

```diff
@@ -161,12 +161,19 @@ const Dropdown: React.FunctionComponent<IDropdownProps & ThemeProps<DefaultTheme
           if (onInputValueChange) {
             if (stateAndHelpers.isOpen) {
               onInputValueChange(inputVal, stateAndHelpers);
-            } else if (dropdownType !== 'combobox') {
+            } else if (dropdownType === 'multiselect') {
               onInputValueChange('', stateAndHelpers);
             }
           }
         }}
         onStateChange={(changes, stateAndHelpers) => {
+          if (
+            dropdownType === 'autocomplete' &&
+            changes.isOpen === false &&
+            !changes.selectedItem
+          ) {
+            onSelect && onSelect(selectedItem, stateAndHelpers);
+          }
```

Comment:
Ensure we go back to the original selection in the event the user removes part of the input but doesn't commit to the change - by blurring or by selecting the same option again (which wouldn't trigger a selection on its own as that won't register as a "change").